CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -956,381 +1460,627 @@ public int DestroyDescriptorPoolResources() {
+            if (cvkVertexFlagsStagingBuffer != null) {
+                cvkVertexFlagsStagingBuffer.Destroy();
+                cvkVertexFlagsStagingBuffer = null;
+            }       
+            
+            if (newVertexCount > 0) {
+                int vertexFlagsBufferSizeBytes = newVertexCount * FLAGS_STRIDE;
+                cvkVertexFlagsStagingBuffer = CVKBuffer.Create(cvkDevice, 
+                                                               vertexFlagsBufferSizeBytes, 
+                                                               VK_BUFFER_USAGE_TRANSFER_SRC_BIT,
+                                                               VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
+                cvkVertexFlagsStagingBuffer.DEBUGNAME = "CVKIconsRenderable.RebuildVertexFlagsStagingBuffers cvkVertexFlagsStagingBuffer";
+                
+                ByteBuffer pVertexFlagsMemory = cvkVertexFlagsStagingBuffer.StartWrite(0, vertexFlagsBufferSizeBytes);
+                for (int pos = 0; pos < newVertexCount; pos++) {
+                    SetVertexFlags(pos, 0, pVertexFlagsMemory, access);
+                }
+                cvkVertexFlagsStagingBuffer.EndWrite();
+                pVertexFlagsMemory = null; // now unmapped, do not use           
+            }
+        } finally {
+            vertexFlagsStagingBufferLock.unlock();
+        }         
+    }
+    
+    public CVKRenderableUpdateTask TaskRebuildVertexFlags(final VisualAccess access) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        final int newVertexCount = access.getVertexCount();
+//        CVKAssert(newVertexCount == vertexCount);
+        CVKLOGGER.info(String.format("TaskRebuildVertexFlags frame %d: %d verts", parent.GetFrameNumber(), access.getVertexCount()));
+        RebuildVertexFlagsStagingBuffers(access, newVertexCount);
+        
+        //=== EXECUTED BY RENDER THREAD (during CVKVisualProcessor.ProcessRenderTasks) ===//
+        return () -> {            
+            vertexFlagsBufferState = CVK_RESOURCE_NEEDS_REBUILD;
+        };     
+    }
+    
+    public CVKRenderableUpdateTask TaskUpdateVertexFlags(final VisualChange change, final VisualAccess access) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        // If this fires investigate why we didn't get a rebuild task first
+        CVKLOGGER.info(String.format("TaskUpdateVertexFlags frame %d: %d verts", parent.GetFrameNumber(), access.getVertexCount()));
+        final int newVertexCount = access.getVertexCount();
+//        CVKAssert(vertexCount != newVertexCount); //REMOVE AFTER TESTING
+        
+        // If we have had an update task called before a rebuild task we first have to build
+        // the staging buffer.  Rebuild also if we our vertex count has somehow changed.
+        final boolean rebuildRequired = cvkVertexFlagsStagingBuffer == null || vertexCount != access.getVertexCount();
```

Comment:
Is it possible for say a single node to be deleted and then a single added? meaning the vertex count would be the same but we would actually need a rebuild?

Reply:
Yes but in that case many of the resource states should get flagged for rebuild.  Once selection is working though we definitely need to test this.